typing/typemod.ml

```diff
@@ -1853,8 +1853,10 @@ and type_module_aux ~alias sttn funct_body anchor env smod =
           let mty_appl =
             match path with
               Some path ->
-                Subst.modtype (Subst.add_module param path Subst.identity)
-                              mty_res
+                let scope = Ctype.create_scope () in
```

Comment:
Why not use the scope of the path of this applicative functor?

Reply:
I think it would have been sound, but seems less correct that creating a new scope, because you'd end up with:
```ocaml
module M = F(A)
```
where the scope of `M` is bigger than the one of the items of the resulting signature.

Reply:
I see. One day, `M` will be handled as a module alias, and then this may make a difference.
But I agree that at this point there seems to be no way to distinguish between the two, and creating a new scope is more in line with the current semantics.